### PR TITLE
XWIKI-22485: Livedata option dropdown has no grouping semantics

### PR DESCRIPTION
**Pull request by Sereza7**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-22485

# Changes

## Description

* Updated the structure of the LiveData dropdown (namely, removed separator <li>s and grouped links together)
* Updated bootstrap dropdown styles to account for this new kind of separator in dropdowns.
* Updated the tests to work on this updated architecture

## Clarifications

* Using separator list items can cause weird experiences for AT users, so instead of this I decided to just nest sublists inside the main list to make a semantic separation between the items.
* I brought the style changes directly on our fork of bootstrap because it made the most sense here. I could have added this style in the liveData code, but this change is generic so I thought it would be best as an update of our design system.
* I decided to keep the changes on the tests minimal. I could have reworked them to make them simpler. The workaround that was used is no longer needed. However, in order to simplify this, I'd need to rewrite a large part of each test because of the way it was coded. I went with the easy solution, feel free to let me know if we should add more changes to the tests in this PR.

# Screenshots & Video

On the left is the dropdown before the changes in this PR, and on the right is the same dropdown after the changes in this PR. We can see that the styles are pretty much the same. This was the goal of this PR: improve the HTML architecture of the dropdown without regressing on the style.
Screenshot from 2024-11-20 12-05-23

# Executed Tests

Successfully built the livedata webjar with the quality profile: `mvn clean install -f xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-test/xwiki-platform-livedata-test-docker/`.
Successfully passed livedata docker tests: `mvn clean install -f xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-test/xwiki-platform-livedata-test-docker/`.
Manual tests on a local instance to check for style with Firefox (see screenshot above).
Sonarlint analysis on the updated files did not point out any recent quality regressions.

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * 16.10.X

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <ul class="dropdown-menu dropdown-menu-right">
 
       <!-- Actions -->
-      <li class="dropdown-header">{{ $t('livedata.dropdownMenu.actions') }}</li>
+      <li><span class="dropdown-header">{{ $t('livedata.dropdownMenu.actions') }}</span><ul>
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue:50]: WDYT of indenting the new `ul` block?

File: xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue [hunk 2]
```diff
@@ -56,12 +56,10 @@
...
         </a>
       </li>
-
+      </ul></li>
 
       <!-- Layouts -->
-      <li role="separator" class="divider"></li>
-
-      <li class="dropdown-header">{{ $t('livedata.dropdownMenu.layouts') }}</li>
+      <li><span class="dropdown-header">{{ $t('livedata.dropdownMenu.layouts') }}</span><ul>
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue:62]: Same question as https://github.com/xwiki/xwiki-platform/pull/3662/files#r1853982643 here too

**Review by manuelleduc — COMMENTED** [review]
Looks good. Remains one code indentation remark.
You don't mention what was wrong/what is improved for users using assistive technologies, it could be useful to understand what you try to achieve in this PR.

**Comment by Sereza7** [comment]
> You don't mention what was wrong/what is improved for users using assistive technologies, it could be useful to understand what you try to achieve in this PR.

I'll add this paragraph directly as a comment on the ticket too for reference sake.
As a user with vision, when you see this dropdown
image
You understand that there's a list of options/ actions, and that this list is split into subcategories. This is the semantics we share through style. 
The semantics we share through structure of the HTML should be similar. Before this PR, it was:
image
A list of options/ actions. We did not convey any information about the groups in the HTML.

For the sake of users of assistive tools, and overall accessibility of XWiki, you want to make sure the semantics you convey through your HTML are roughly the same as the semantics you convey through your style. This is [adaptability](https://www.w3.org/WAI/fundamentals/accessibility-principles/#adaptable).

This PR makes sure we represent those subcategories in the HTML structure. Additionally, it removes the "I am one item of the list" semantics from the separator, because a separator is not an item anyone would name when asked to list the elements of this dropdown (meaning it has no semantics and should be showed using non semantic elements/ no element style only).



____
Before AT would approx read:
"List with 6 items - item 1:Actions - item 2: Refresh - item 3:separator - item 4: Layouts - item 5:Table - item6: Cards [...]"
With the new structure they would read:
"List with 2 items - item 1: Actions, List with 1 item - item 1: Refresh - item 2: Layouts, List with 2 items - item 1: Table - item 2: Cards"

File: xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue [hunk 3]
```diff
@@ -47,47 +47,42 @@
...
-
-      <li class="dropdown-header">{{ $t('livedata.dropdownMenu.panels') }}</li>
-
-      <li v-for="panel in logic.panels" :key="panel.id">
-        <a href="#" @click.prevent="logic.uniqueArrayToggle(logic.openedPanels, panel.id)">
-          <XWikiIcon :icon-descriptor="{name: panel.icon}"/>
-          {{ panel.name }}
-        </a>
-      </li>
+      <li><span class="dropdown-header">{{ $t('livedata.dropdownMenu.panels') }}</span><ul>
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue:78]: that part could be indented as well.

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/LivedataDropdownMenu.vue]: Addressed in https://github.com/xwiki/xwiki-platform/pull/3662/commits/3cb1436797723d677f033e7c7000b38115de3d87 :+1:

**Pull request merged**